tests/test_datetime_transformer.py

```diff
@@ -0,0 +1,192 @@
+    assert transformer.variables == ["var_A", "var_D"]
+    # transform
+    assert transformer.features_to_add == {
+        "var_A": list(transformer.feature_map.keys()),
+        "var_D": list(transformer.feature_map.keys())
+    }
+    assert transformer.keep_original is False
+    pd.testing.assert_frame_equal(X, df, check_like=True)
+
+
+def test_datetime_select_dict_specific(dataframe_datetime_multiple):
+    transformer = DateTimeTransformer(
+        variables=["var_A", "var_D"],
+        features_to_add={
+            "var_A": ["year", "month", "day"],
+            "var_D": ["is_weekend", "semester", "hr", "min", "sec"]
+        }
+    )
+    X = transformer.fit_transform(dataframe_datetime_multiple)
+
+    # expected
+    df = pd.DataFrame({
+        "var_A_year": [2009, 2010, 2001],
+        "var_A_month": [1, 8, 11],
+        "var_A_day": [5, 9, 18],
+        "var_B": ["John", "Jason", "Ethan"],
+        "var_C": [25, 28, 30],
+        "var_D_semester": [1, 2, 2],
+        "var_D_is_weekend": [1, 1, 1],
+        "var_D_hr": [6, 12, 13],
+        "var_D_min": [5, 5, 1],
+        "var_D_sec": [59, 21, 36],
+
+    })
+
+    # init parameters
+    assert transformer.variables == ["var_A", "var_D"]
+    # transform
+    assert transformer.features_to_add == {
+        "var_A": ["year", "month", "day"],
+        "var_D": ["is_weekend", "semester", "hr", "min", "sec"]
+    }
+    assert transformer.keep_original is False
+    pd.testing.assert_frame_equal(X, df, check_like=True)
+
+
+def test_datetime_select_non_existent_columns(dataframe_datetime_multiple):
+    with pytest.raises(KeyError):
+        transformer = DateTimeTransformer(
+            variables=["var_X"]
+        )
+        X = transformer.fit_transform(dataframe_datetime_multiple)
+
+    with pytest.raises(ValueError):
+        transformer = DateTimeTransformer(
+            variables=["var_A"],
+            features_to_add={
+                "var_X": ["year"]
+            }
+        )
+        X = transformer.fit_transform(dataframe_datetime_multiple)
+
+
+def test_datetime_non_existent_features(dataframe_datetime_normal):
+    with pytest.raises(ValueError):
+        transformer = DateTimeTransformer(
+            variables=["var_A"],
+            features_to_add=["year", "month", "nanosecond"]
+        )
+        X = transformer.fit_transform(dataframe_datetime_normal)
+
+    with pytest.raises(ValueError):
+        transformer = DateTimeTransformer(
+            variables=["var_A"],
+            features_to_add={
+                "var_A": ["year", "month", "nanosecond"]
+        })
+        X = transformer.fit_transform(dataframe_datetime_normal)
+
+
+def test_datetime_invalid_error(dataframe_datetime_invalid):
+    with pytest.raises(ValueError):
+        transformer = DateTimeTransformer(
+            variables=["var_A"],
+            features_to_add=["year", "month", "day"],
+        )
+        X = transformer.fit_transform(dataframe_datetime_invalid)
+
+
```

Comment:
can we test situations where the datetime variable only contains the date fraction or the time fraction? and the user indicates for example all features by default? there should be a warning there that this is not possible (to include in the main transformer). 
And then also test the cases where it could work, so if datetime contains only time, and user asks min, hr, sec, it should work. If datetime contains only date, and user asks day, month, etc, it should also work.